daemon/networkdriver/bridge/driver.go

```diff
@@ -242,18 +355,48 @@ func InitDriver(config *Config) error {
 
 	if config.EnableIpForward {
 		// Enable IPv4 forwarding
-		if err := ioutil.WriteFile("/proc/sys/net/ipv4/ip_forward", []byte{'1', '\n'}, 0644); err != nil {
-			logrus.Warnf("WARNING: unable to enable IPv4 forwarding: %s\n", err)
+		if err := setKernelBoolParam(getForwardingKernelParam(ipv4, ""), true); err != nil {
+			logrus.Warnf("Unable to enable IPv4 forwarding: %v", err)
 		}
 
 		if config.FixedCIDRv6 != "" {
 			// Enable IPv6 forwarding
-			if err := ioutil.WriteFile("/proc/sys/net/ipv6/conf/default/forwarding", []byte{'1', '\n'}, 0644); err != nil {
-				logrus.Warnf("WARNING: unable to enable IPv6 default forwarding: %s\n", err)
+			if err := setKernelBoolParam(getForwardingKernelParam(ipv6, ""), true); err != nil {
+				logrus.Warnf("Unable to enable IPv6 default forwarding: %v", err)
+			}
+			if err := setKernelBoolParam(getForwardingKernelParam(ipv6, "all"), true); err != nil {
+				logrus.Warnf("Unable to enable IPv6 all forwarding: %v", err)
+			}
+		}
+	}
+
+	if !config.InterContainerCommunication && config.EnableIptables {
+		ipVersion := ipv4
+		tryLoad := false
+		if config.FixedCIDR != "" || config.EnableIPv6 {
+			ipVersion |= ipv6
+		}
+		err := enableBridgeNetFiltering(ipVersion, bridgeIface)
+		if err != nil {
+			if ptherr, ok := err.(*os.PathError); ok {
+				if errno, ok := ptherr.Err.(syscall.Errno); ok && errno == syscall.ENOENT {
+					tryLoad = true
+				}
 			}
-			if err := ioutil.WriteFile("/proc/sys/net/ipv6/conf/all/forwarding", []byte{'1', '\n'}, 0644); err != nil {
-				logrus.Warnf("WARNING: unable to enable IPv6 all forwarding: %s\n", err)
+		}
+		if tryLoad {
```

Comment:
I don't understand why we need this block, it is too late here to load module. You should just throw error or load module before enabling filtering.